### PR TITLE
NeMo 2.0 In-framework deployment support

### PR DESCRIPTION
**Pull request by oyilmaz-nvidia**
# What does this PR do ?

Adds nemo2 support for inframework deployment

### CODE REVIEW

File: docs/source/tts/data/ngc_models_codec.csv [hunk 1]
```diff
@@ -4,4 +4,3 @@ mel_codec_22khz_medium,LibriVox and Common Voice,22050Hz,nemo.collections.tts.mo
...
 mel_codec_22khz_fullband_medium,LibriVox and Common Voice,22050Hz,nemo.collections.tts.models.AudioCodecModel,`mel_codec_22khz_fullband_medium <https://ngc.nvidia.com/catalog/models/nvidia:nemo:mel_codec_22khz_fullband_medium>`_,``https://api.ngc.nvidia.com/v2/models/nvidia/nemo/mel_codec_22khz_fullband_medium/versions/v1/files/mel_codec_22khz_fullband_medium.nemo``
 mel_codec_44khz_fullband_medium,LibriVox and Common Voice,44100Hz,nemo.collections.tts.models.AudioCodecModel,`mel_codec_44khz_fullband_medium <https://ngc.nvidia.com/catalog/models/nvidia:nemo:mel_codec_44khz_fullband_medium>`_,``https://api.ngc.nvidia.com/v2/models/nvidia/nemo/mel_codec_44khz_fullband_medium/versions/v1/files/mel_codec_44khz_fullband_medium.nemo``
-nvidia/low-frame-rate-speech-codec-22khz,LibriVox and Common Voice,22050Hz,nemo.collections.tts.models.AudioCodecModel,`audio_codec_low_frame_rate_22khz <https://huggingface.co/nvidia/low-frame-rate-speech-codec-22khz>`_,``https://huggingface.co/nvidia/low-frame-rate-speech-codec-22khz/resolve/main/low-frame-rate-speech-codec-22khz.nemo``
```
Comment (athitten) [inline docs/source/tts/data/ngc_models_codec.csv:7]: @oyilmaz-nvidia just checking if this is intended or deleted by mistake since looks unrelated to the scope of this PR ?

**Review by athitten — COMMENTED** [review]
Thanks @oyilmaz-nvidia LGTM! Tested locally and worked as expected. There's just a minor comment.

File: docs/source/tts/checkpoints.rst [hunk 2]
```diff
@@ -156,6 +156,6 @@ End2End models
 Codec models
 ^^^^^^^^^^^^
 .. csv-table::
-   :file: data/models_codec.csv
+   :file: data/ngc_models_codec.csv
    :align: left
```
Comment (athitten) [inline docs/source/tts/checkpoints.rst:160]: Also curious if this is intended ? Just making sure

File: docs/source/tts/data/ngc_models_codec.csv (hunk 1 above)
Comment (oyilmaz-nvidia) [reply docs/source/tts/data/ngc_models_codec.csv:7]: Pulled the @hemildesai 's PR and I guess this is coming from his PR.

File: docs/source/tts/checkpoints.rst (hunk 2 above)
Comment (oyilmaz-nvidia) [reply docs/source/tts/checkpoints.rst:160]: Pulled the @hemildesai 's PR and I guess this is coming from his PR.
